pkg/ccl/streamingccl/streampb/stream.proto

```diff
@@ -42,6 +44,26 @@ message StreamPartitionSpec {
   ExecutionConfig config = 3 [(gogoproto.nullable) = false];
 }
 
+message ReplicationStreamSpec {
+  message Partition {
+    // ID of the node this partition resides
+    int32 node_id = 1 [(gogoproto.customname) = "NodeID",
+      (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/roachpb.NodeID"];
+
+    // The SQL address of the node.
+    util.UnresolvedAddr sql_address = 2 [(gogoproto.nullable) = false,
+      (gogoproto.customname) = "SQLAddress"];
+
+    // Locality of the node
+    roachpb.Locality locality = 3 [(gogoproto.nullable) = false];
+
+    // The spec of the processor responsible for streaming this partition
+    StreamPartitionSpec partition_spec = 4 [(gogoproto.customname) = "PartitionSpec"];
+  }
+
+  repeated Partition partitions = 1;
```

Comment:
We don't have nil elements in this list, right, or use nil vs not-nil Partition anywhere? should this be `[(gogoproto.nullable) = false];` ?